odoo/addons/base/models/ir_binary.py

```diff
@@ -45,15 +46,12 @@ def _find_record(
             record = self.env[res_model].browse(res_id).exists()
         if not record:
             raise MissingError(f"No record found for xmlid={xmlid}, res_model={res_model}, id={res_id}")
-
-        record = self._find_record_check_access(record, access_token, field)
-        return record
-
-    def _find_record_check_access(self, record, access_token, field):
-        if record._name == 'ir.attachment':
-            return record.validate_access(access_token)
-
-        record.check_access('read')
+        try:
+            record.check_access("read")
+        except AccessError:
+            if record._can_elevate_access(field, access_token):
+                return record.sudo()
```

Comment:
What a lousy method name.  Out of context, you have no clue what the method is for.
Please make the method name more specific, like `_can_return_content()` or `_can_export_content()`, so that one can connect the method to the controllers `/web/content` and `/web/image`.